clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -179,7 +193,8 @@ def CIR_LongDouble : CIR_FloatType<"LongDouble", "long_double"> {
 
 // Constraints
 
-def CIR_AnyFloat: AnyTypeOf<[CIR_Single, CIR_Double, CIR_LongDouble]>;
+def CIR_AnyFloat: AnyTypeOf<[CIR_Single, CIR_Double, CIR_FP16, CIR_BFloat16, 
```

Comment:
Do all operations currently using `CIR_AnyFloat` support 16-bit fp?

Reply:
I believe yes. The operations currently using `CIR_AnyFloat` include:

- Builtin fp2fp unary operations family, e.g. `cir.ceil`, `cir.cos`, `cir.exp`, etc.
- Atomic fetch operation.

All operations above should support `f16` and `bf16`. See https://godbolt.org/z/rq5edMarM .